src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -125,7 +125,12 @@ const DeleteWorkspaceModal = ajaxCaller(class DeleteWorkspaceModal extends Compo
     try {
       this.setState({ deleting: true })
       await Workspaces.workspace(namespace, name).delete()
-      Nav.goToPath('workspaces')
+      console.log(Nav.history.location.pathname)
+      if (Nav.history.location.pathname !== '/workspaces') {
+        Nav.goToPath('workspaces')
+      } else {
+        document.location.reload()
```

Comment:
Rather than encode the behavior with an `if`, it might be better to add a callback prop so each consumer can implement their own action. As a concrete benefit, that will allow us to do a soft refresh of the workspaces list instead of reloading the whole page.